src/__tests__/ParserComponent.test.tsx

```diff
@@ -61,22 +62,214 @@ describe('Csv FileEvents processed correctly', () => {
+      //   console.log(error);
+      // }
+    });
+
+    it('.js rejected', async () => {
+      const jsTestFile: File = new File(
+          ['abcdefg'],
+          'test.js',
+          {type: '.js, text/javascript'},
+      );
+      const fileEvent: any = {target: {files: [jsTestFile]}};
+      // try {
+      //   // todo: make sure it throws the right error
+      //   expect(await comp.instance().parse(fileEvent)).toThrowError();
+      // } catch (error) {
+      //   console.log(error);
+      // }
     });
   });
 
-  describe('componentDidMount()', () => {
-    it('dummy test', () => {
-      // todo: devs need to write unit tests
+  it('.csv file with no temporal data is rejected', async () => {
+    const onChangeMock = jest.fn();
+    const noDateFile: File = new File(
+        ['H1,H2,H3,H4\n' +
+                'a,b,c,d\n' +
+                'e,f,g,h\n' +
+                'i,j,k,l\n'],
+        'test.csv',
+        {type: '.csv,text/csv'},);
+    const comp: any = mount(<ParserComponent
+      {...props}
+      onChange={onChangeMock}
+    />);
+    const fileEvent = {target: {files: [noDateFile]}};
+    try {
+      expect(
+          await comp.instance().parse(fileEvent)
+      ).toThrowError('The file uploaded has no dates.');
+    } catch (error) {
+      console.log(error);
+    }
+
+    const fileUsed: File = onChangeMock.mock.calls[0][0];
+    expect(onChangeMock).toHaveBeenCalledTimes(1);
+    expect(fileUsed.name).toBe(noDateFile.name);
+  });
+
+  describe('.csv with different date formats processed correctly', () => {
+    const onChangeMock = jest.fn();
+    const comp: any = mount(<ParserComponent
+      {...props}
+      onChange={onChangeMock}
+    />);
+    let compData: Array<object>;
+
+    beforeEach(() => {
+      onChangeMock.mockClear();
+      comp.setState({data: []});
+      compData = comp.state('data');
+      expect(compData.length).toBe(0);
+    });
+
+    afterEach(() => {
+      // onChange should be called once, and it should get to parseCsv()
+      expect(onChangeMock).toHaveBeenCalledTimes(1);
+      expect(compData.length).toBe(3);
+    });
+
+    it('.csv with sorted dates accepted', async () => {
+      const multiDateFile: File = new File(
+          // todo: add more date formats
+          ['Date,SomeNum,SomeString\n' +
+                  '04/04/1995,4,abcd\n' +
+                  '06-07-1996,5,efg\n' +
+                  'November 5 1997,1,hij\n' +
+                  ''],
+          'multiDateTest.csv',
+          {type: '.csv,text/csv'},
+      );
+      const fileEvent = {target: {files: [multiDateFile]}};
+
+      // no error should be thrown!!!
+      try {
+        expect(await comp.instance().parse(fileEvent)).toBe(undefined);
+      } catch (error) {
+        fail(); // fail if error thrown
+      }
+      // data should be updated to contain csv info
+      compData = comp.state('data');
+      // Check that the object contains all the data from the csv
```

Comment:
But it doesn't really check that right? It just sees that it turned into "some number"? What if it incorrectly parses the dates?